ecosystem/sep-0031.md

```diff
@@ -153,8 +153,25 @@ The response should be a JSON object like:
          "fee_percent":1,
          "min_amount":0.1,
          "max_amount":1000,
-         "sender_sep12_type": "sep31-sender",
-         "receiver_sep12_type": "sep31-receiver",
+         "sender_sep12_required": true,
+         "receiver_sep12_required": true,
+         "sep12_types": {
+          "sep31-sender": {
```

Comment:
🤔 How does someone go from knowing that `sender_sep12_required` is `true`, to needing to send the `sep31-sender` type? How do they know not to use `sep31-receiver`? How does an implementer communicate that KYC is required for amounts over a certain amount, but not required for under? Or is that not relevant for this SEP?

In the cases that the `_required` fields are `false`, would there still be optional types defined for them, or there would be no types?

If there are two types of SEP-12 types for SEP-31, maybe it is worth separating them so discovering which to use doesn't require parsing the string names and looking for `-sender` and `-receiver` strings.

For example:
```javascript
{
   "receive":{
      "USD":{
        "sender_sep12_types": {
          "sep31-sender": {
            "description": "U.S. citizens limited to sending payments of less than $10,000 in value"
          },
          "sep31-large-sender": {
            "description": "U.S. citizens that do not have sending limits"
          },
          "sep31-foreign-sender": {
            "description": "non-U.S. citizens sending payments of less than $10,000 in value"
          }
        },
        "receiver_sep12_types": {
          "sep31-receiver": {
            "description": "U.S. citizens receiving USD"
          },
          "sep31-foreign-receiver": {
            "description": "non-U.S. citizens receiving USD"
          }
        },
        // ...
      }
   }
}
```

Reply:
> In the cases that the _required fields are false, would there still be optional types defined for them, or there would be no types?

There would be no types. I don't see a use case for optionally KYC'ing a customer.

> If there are two types of SEP-12 types for SEP-31, maybe it is worth separating them so discovering which to use doesn't require parsing the string names and looking for `-sender` and `-receiver` strings.

I agree, I'll make that change. Using `-sender` and `-receiver` suffixes is just a pattern I've developed too -- the `type` values defined here could potentially not follow that pattern. 

> How does an implementer communicate that KYC is required for amounts over a certain amount, but not required for under? Or is that not relevant for this SEP?

Ohh thats a good question, that could be a possibility. The first thing that comes to mind is to define types for all customers -- even those that don't need KYC -- and return an empty `fields` object from `GET /customer` for that no-KYC-necessary type. I can't think of a better way at the moment. The other option is to keep the `_required` field `true` and simply omit a type that would apply to the customer's situation, but that is more confusing than my first suggestion.

Reply:
I added a statement to convey how to handle that case.